versions/1.21.5/src/main/java/at/hannibal2/skyhanni/features/foraging/InvisibugHighlighter.kt

```diff
@@ -3,60 +3,48 @@ package at.hannibal2.skyhanni.features.foraging
 import at.hannibal2.skyhanni.SkyHanniMod
 import at.hannibal2.skyhanni.api.event.HandleEvent
 import at.hannibal2.skyhanni.data.IslandType
-import at.hannibal2.skyhanni.data.IslandTypeTags
-import at.hannibal2.skyhanni.events.MobEvent
 import at.hannibal2.skyhanni.events.ReceiveParticleEvent
 import at.hannibal2.skyhanni.events.minecraft.SkyHanniRenderWorldEvent
-import at.hannibal2.skyhanni.mixins.hooks.RenderLivingEntityHelper
+import at.hannibal2.skyhanni.events.minecraft.SkyHanniTickEvent
 import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
-import at.hannibal2.skyhanni.utils.ChatUtils
 import at.hannibal2.skyhanni.utils.ColorUtils.toColor
 import at.hannibal2.skyhanni.utils.EntityUtils
+import at.hannibal2.skyhanni.utils.EntityUtils.canBeSeen
 import at.hannibal2.skyhanni.utils.LocationUtils.distanceTo
-import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
-import at.hannibal2.skyhanni.utils.LorenzVec
-import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.collection.CollectionUtils.removeNotContainedFromList
 import at.hannibal2.skyhanni.utils.getLorenzVec
 import at.hannibal2.skyhanni.utils.render.WorldRenderUtils.drawWaypointFilled
 import net.minecraft.entity.LivingEntity
 import net.minecraft.entity.decoration.ArmorStandEntity
 import net.minecraft.particle.ParticleTypes
-import kotlin.time.Duration.Companion.seconds
 
 @SkyHanniModule
 object InvisibugHighlighter {
     val config get() = SkyHanniMod.feature.foraging.foragingMobHighlight.invisibugHighlight
 
-    private fun isEnabled() = config.enabled && isInIsland()
-    private fun isInIsland() = IslandTypeTags.FORAGING_CUSTOM_TREES.inAny()
-
-    private val particlesToHighlight = mutableMapOf<LorenzVec, SimpleTimeMark>()
-    private val invisibugEntities = mutableSetOf<LivingEntity>()
+    private val invisibugEntities = mutableListOf<LivingEntity>()
 
     @HandleEvent(onlyOnIsland = IslandType.GALATEA)
     fun onParticle(event: ReceiveParticleEvent) {
+        if (!config.enabled) return
+
         val particle = event.type
         if (particle != ParticleTypes.CRIT) return
 
-        val nearestArmorStand = EntityUtils.getAllEntities()
-            .filterIsInstance<ArmorStandEntity>()
+        val nearestArmorStand = EntityUtils.getEntitiesNearby<ArmorStandEntity>(event.location, 5.0)
             .minByOrNull { it.distanceTo(event.location) }
 
-        if (nearestArmorStand == null || nearestArmorStand.distanceTo(event.location) > 5) return
+        if (nearestArmorStand == null) return
```

Comment:
Add an is default check here, as you only look for armorstands that have the default name (looked at the shcopyentitys you send in contributer chat)
```suggestion
        if (nearestArmorStand == null || !nearestArmorStand.isDefaultValue()) return
```